### PR TITLE
feat(app): build dictionary app

### PR DESCRIPTION
**Pull request by ellaboevans**
This pull request contains the solution for the Dictionary app

### CODE REVIEW

**Comment by fredrickyalley** [comment]
The toggle should have some tooltip or hover message for dark/white theme

File: week_3/dictionary-app/css/components/_main-content.scss [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    }
+  }
+}
+
+.class--heading {
+  display: flex;
+  align-items: center;
+  gap: 1.5rem;
+}
+.searched-word__category {
```
Comment (fredrickyalley) [inline week_3/dictionary-app/css/components/_main-content.scss:54]: the style class should have some consistency with the naming. use _ or - or any other style but keep it consistent

**Comment by fredrickyalley** [comment]
there is a scroll on the page if the word doesn't exist. fix that

**Comment by fredrickyalley** [comment]
if you open the app, u shouldn't see keyboard definition because, with the screen you were provided, it was what was been searched for

File: week_3/dictionary-app/css/components/_main-content.scss (hunk 1 above)
Comment (ellaboevans) [reply week_3/dictionary-app/css/components/_main-content.scss:54]: Yes please, I will do exactly that.

File: week_3/dictionary-app/scripts/main.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  setTimeout(() => {
+    searchInput.style.outline = "";
+    inputErrorMessage.style.visibility = "hidden";
+  }, 2000);
+}
+
+// Event listeners
+searchButton.addEventListener("click", () => {
+  const word = searchInput.value.trim();
+  if (word) {
```
[the remark this answers is not in the document]
Comment (ellaboevans) [reply week_3/dictionary-app/scripts/main.js:116]: Sure, I will do that as well.

**Comment by ellaboevans** [comment]
> there is a scroll on the page if the word doesn't exist. fix that

This has been fixed.

**Comment by ellaboevans** [comment]
> The toggle should have some tooltip or hover message for dark/white theme

Tooltip has successfully been added to tip the dark and light them mode value

File: week_3/dictionary-app/scripts/main.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ellaboevans) [reply week_3/dictionary-app/scripts/main.js:116]: I have added the length validation

File: week_3/dictionary-app/scripts/main.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  } catch (error) {
+    displayErrorMessage();
+  } finally {
+    loader.style.display = "none";
+    overlay.style.display = "none";
+  }
+}
+
+function displayInitialMessage() {
+  let p = document.createElement("p");
```
Comment (fredrickyalley) [inline week_3/dictionary-app/scripts/main.js:61]: name if paragragh, p is not descriptive enough

Comment (ellaboevans) [reply week_3/dictionary-app/scripts/main.js:61]: Done!

**Review by fredrickyalley — APPROVED** [review]

**Pull request merged**